### PR TITLE
Nv 2517 display that workflow has nodes that will fail to send a notification

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?

- get active integrations status returns an object which contains 

 `{channels : {channelType: boolean} }` which describes if there's any active integrations for the particular channelType
 along with and `isActive: boolean` field which denotes if all the steps of a single workflow have active integrations or not
### Other information (Screenshots)

image

### CODE REVIEW

File: apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        ...workflows,
+        activeIntegrationStatus: {
+          channels: activeChannelsStatus,
+          isActive,
+        },
+      };
+    }
+  }
+
+  private handleSteps(steps: NotificationStep[], activeChannelsStatus: { [key in ChannelTypeEnum]: boolean }) {
```
Comment (BiswaViraj) [inline apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts:78]: loops through the steps and updates the `isActive` flag if the step has a channelType which doesn't have any active integrations.

File: apps/api/src/app/workflows/dto/workflow-response.dto.ts [hunk 2]
```diff
@@ -115,4 +116,13 @@ export class WorkflowResponse {
   @ApiPropertyOptional()
   @IsOptional()
   data?: NotificationTemplateCustomData;
+  activeIntegrationStatus?: ActiveIntegrationsChannelsStatus;
+
+  @ApiProperty()
+  isBlueprint: boolean;
```
Comment (davidsoderberg) [inline apps/api/src/app/workflows/dto/workflow-response.dto.ts:122]: Should we really expose these in the swagger json? right now this is only used internally for use?

File: apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    const activeIntegrations = await this.getActiveIntegrationUsecase.execute(
+      GetActiveIntegrationsCommand.create({
+        organizationId: command.organizationId,
+        environmentId: command.environmentId,
+        userId: command.userId,
+      })
+    );
+
+    const activeIntegrationsByEnv = activeIntegrations.filter(
```
[the remark this answers is not in the document]
Comment (davidsoderberg) [reply apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts:31]: Why do we need to do this when we pass environment id here: https://github.com/novuhq/novu/pull/3966/files#diff-dcd9b18e319e2a1e95ddca88446db6525e43dea143d7aaca024d6e563cbafc3fR26

File: apps/api/src/app/workflows/dto/workflow-response.dto.ts (hunk 2 above)
Comment (BiswaViraj) [reply apps/api/src/app/workflows/dto/workflow-response.dto.ts:122]: Agree, we shouldn't expose it, let me remove it

File: apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BiswaViraj) [reply apps/api/src/app/workflows/usecases/get-active-integrations-status/get-active-integrations-status.usecase.ts:31]: @davidsoderberg
We do not use environmentId if the multiProvider is enabled
https://github.com/novuhq/novu/blob/f86886414ebb8853e9f83c11e06afacb3aa52864/packages/application-generic/src/usecases/get-decrypted-integrations/get-decrypted-integrations.usecase.ts#L31

**Review by davidsoderberg — APPROVED** [review]

**Pull request merged**